src/close-test.js

```diff
@@ -4,154 +4,94 @@
+
+      const s1 = dialerMuxer.newStream()
+      const s2 = dialerMuxer.newStream()
+
+      // close the listener in a bit
+      setTimeout(() => tcpListener.close(), 50)
+
+      // test is complete when all muxed streams have closed with error
+      await Promise.all([s1, s2].map(stream => {
+        return (async () => {
+          try {
+            await pipe(infiniteRandom, stream, consume)
+          } catch (err) {
+            return expect(err).to.exist.mark()
+          }
+          throw new Error('stream did not throw')
+        })()
+      }))
     })
 
     it('closing one of the muxed streams doesn\'t close others', (done) => {
       const p = pair()
-      const dialer = muxer.dialer(p[0])
-      const listener = muxer.listener(p[1])
-
-      expect(6).checks(done)
-
-      const conns = []
-
-      listener.on('stream', (stream) => {
+      const dialer = new Muxer()
+      const listener = new Muxer(stream => {
         expect(stream).to.exist.mark()
-        pull(stream, stream)
+        pipe(stream, stream)
       })
 
-      for (let i = 0; i < 5; i++) {
-        conns.push(dialer.newStream())
-      }
-
-      conns.forEach((conn, i) => {
-        if (i === 1) {
-          closeAndWait(conn, (err) => {
-            expect(err).to.not.exist.mark()
-          })
-        } else {
-          pull(
-            conn,
-            pull.onEnd(() => {
-              throw new Error('should not end')
-            })
-          )
-        }
-      })
-    })
+      pipe(p[0], dialer, p[0])
+      pipe(p[1], listener, p[1])
 
-    it.skip('closing on spdy doesn\'t close until all the streams that are being muxed are closed', (done) => {
-      const p = pair()
-      const dialer = muxer.dial(p[0])
-      const listener = muxer.listen(p[1])
+      expect(2).checks(done)
 
-      expect(15).checks(done)
+      const streams = []
 
-      const conns = []
-      const count = []
       for (let i = 0; i < 5; i++) {
-        count.push(i)
+        streams.push(dialer.newStream())
       }
 
-      series(count.map((i) => (cb) => {
-        parallel([
-          (cb) => listener.once('stream', (stream) => {
-            expect(stream).to.exist.mark()
-            pull(stream, stream)
-            cb()
-          }),
-          (cb) => conns.push(dialer.newStream(cb))
-        ], cb)
-      }), (err) => {
-        if (err) return done(err)
-
-        conns.forEach((conn, i) => {
-          pull(
-            pull.values([Buffer.from('hello')]),
-            pull.asyncMap((val, cb) => {
-              setTimeout(() => {
-                cb(null, val)
-              }, i * 10)
-            }),
-            conn,
-            pull.collect((err, data) => {
-              expect(err).to.not.exist.mark()
-              expect(data).to.be.eql([Buffer.from('hello')]).mark()
-            })
-          )
-        })
+      closeAndWait(streams[0])
+
+      streams.slice(1).forEach(async stream => {
```

Comment:
Only 1 of these is actually going to send anything right? As soon as the first iteration happens the `onStream` function will get called and the 2nd expect.mark() will happen, ending the test immediately. 

Since the streams will likely open lazily, shouldn't we start the infinite streams for all of them, then await the close on the first stream? If we don't actually open the other streams first we may miss closures happening.